### PR TITLE
[10637] Restore smart default logging for multiconfig generators

### PR DESCRIPTION
**Pull request by MiguelBarro**
Prior to #1679 the default library behavior was:
- if not using **INTERNAL_DEBUG** never queue infos.
- if using **INTERNAL_DEBUG** on Debug config infos are queued but other configs don't queue them.
- if user defines **LOG_NO_INFO** no infos are queued regardless config or cmake options. 

For multiconfig generators like MSVC this was optimal because Release builds never queued infos and Debug builds may do it or not depending on **INTERNAL_DEBUG** or **LOG_NO_INFO** settings.

In order to allow testing to rely on logging regardless of configuration #1679 introduced new CMake options (**LOG_NO_INFO**, **LOG_NO_WARNING**, **LOG_NO_ERROR**). 

In order to enable info logging on debug builds **LOG_NO_INFO** option was set if **CMAKE_BUILD_TYPE** was debug.  Unfortunately this broke multiconfig generators usual behaviour because:
- they don't use **CMAKE_BUILD_TYPE** to specify the config (but is specified in a second build step by CMake cli).
- Fast-DDS defaults to **CMAKE_BUILD_TYPE=Release** if the variable is not provided.
thus: 
- multiconfig build never log infos by default.
- if logging was enforced by using **LOG_NO_INFO=OFF** then ALL configurations log ruining performance in Release.

To solve this issue:
 - **LOG_NO_INFO** falls back to its previous behavior. If option **LOG_NO_INFO=OFF** it doesn't force all configs to log infos but only debug if **INTERNAL_DEBUG=ON**.
 - **LOG_NO_INFO** fulfillment can be enforced using a new option **FASTDDS_ENFORCE_LOG_INFO**. This reproduces #1679 option behavior and can be used for testing purposes.

### CODE REVIEW

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/3838/)
* Linux aarch64[Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux%20aarch64/996/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/3384/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/3864/)

**Comment by MiguelBarro** [comment]
Verified that commit **685a10d** solves build issues on raspbian stretch.

File: include/fastdds/dds/log/Log.hpp [hunk 1]
```diff
@@ -288,7 +288,10 @@ class LogConsumer
 #define logWarning_(cat, msg)
 #endif // ifndef LOG_NO_WARNING
 
-#if !HAVE_LOG_NO_INFO
+// Allow multiconfig platforms like windows to disable info queueing on Release and other non-debug configs
+#if !HAVE_LOG_NO_INFO &&  \
+    ( defined(ENFORCE_LOG_INFO) || \
```
Comment (MiguelCompany) [inline include/fastdds/dds/log/Log.hpp:293]:
```suggestion
    (defined(ENFORCE_LOG_INFO) || \
```

**Review by MiguelCompany — CHANGES_REQUESTED** [review]
In order to ease integration with other products, I would suggest renaming both the `#define` and the cmake option, prefixing them with `FASTDDS_`

**Review by MiguelCompany — APPROVED** [review]

**Comment by MiguelCompany** [comment]
@richiprosima Please test windows

**Comment by MiguelCompany** [comment]
@richiprosima Please test this

**Comment by MiguelCompany** [comment]
@richiprosima Please test mac

**Comment by MiguelCompany** [comment]
@richiprosima please test this

**Pull request merged**